dissect/target/plugins/apps/browser/chromium.py

```diff
@@ -371,6 +379,127 @@ def extensions(self, browser_name: Optional[str] = None) -> Iterator[BrowserExte
                 except (AttributeError, KeyError) as e:
                     self.target.log.info("No browser extensions found in: %s", json_file, exc_info=e)
 
+    def _get_local_state_key(self, local_state_path: TargetPath) -> Optional[bytes]:
+        """Get the Chromium ``os_crypt`` ``encrypted_key`` and decrypt it using DPAPI."""
+
+        if not local_state_path.exists():
+            self.target.log.warning("File %s does not exist.", local_state_path)
+            return None
+
+        try:
+            local_state_conf = json.loads(local_state_path.read_text())
+        except json.JSONDecodeError:
+            self.target.log.warning("File %s does not contain valid JSON.", local_state_path)
+            return None
+
+        if "os_crypt" not in local_state_conf:
+            self.target.log.warning(
+                "File %s does not contain os_crypt, Chrome is likely older than v80.", local_state_path
+            )
+            return None
+
+        encrypted_key = base64.b64decode(local_state_conf["os_crypt"]["encrypted_key"])[5:]
+        decrypted_key = self.target.dpapi.decrypt_blob(encrypted_key)
+        return decrypted_key
+
+    def passwords(self, browser_name: str = None) -> Iterator[BrowserPasswordRecord]:
+        """Return browser password records from Chromium browsers.
+
+        Chromium on Linux has ``basic``, ``gnome`` and ``kwallet`` methods for password storage:
+            - ``basic`` ciphertext prefixed with ``v10`` and encrypted with hard coded parameters.
+            - ``gnome`` and ``kwallet`` ciphertext prefixed with ``v11`` which is not implemented (yet).
+
+        Chromium on Windows uses DPAPI user encryption.
+
+        The SHA1 hash of the user's password or the plaintext password is required to decrypt passwords
+        when dealing with encrypted passwords created with Chromium v80 (February 2020) and newer.
+
+        You can supply a SHA1 hash or plaintext password using the keychain.
```

Comment:
Is this a Chrome thing or a DPAPI thing? Asking so that I can provide better code suggestions.